drivers/network/dd/e1000/info.c

```diff
@@ -0,0 +1,350 @@
+/*
+ * PROJECT:     ReactOS Intel PRO/1000 Driver
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     Miniport information callbacks
+ * COPYRIGHT:   2013 Cameron Gutman (cameron.gutman@reactos.org)
+ *              2018 Mark Jansen (mark.jansen@reactos.org)
+ */
+
+#include "nic.h"
+
+#include <debug.h>
+
+static ULONG SupportedOidList[] =
+{
+    OID_GEN_SUPPORTED_LIST,
+    OID_GEN_CURRENT_PACKET_FILTER,
+    OID_GEN_HARDWARE_STATUS,
+    OID_GEN_MEDIA_SUPPORTED,
+    OID_GEN_MEDIA_IN_USE,
+    OID_GEN_MAXIMUM_LOOKAHEAD,
+    OID_GEN_MAXIMUM_FRAME_SIZE,
+    OID_GEN_MAXIMUM_SEND_PACKETS,
+    OID_GEN_LINK_SPEED,
+    OID_GEN_TRANSMIT_BUFFER_SPACE,
+    OID_GEN_RECEIVE_BUFFER_SPACE,
+    OID_GEN_RECEIVE_BLOCK_SIZE,
+    OID_GEN_TRANSMIT_BLOCK_SIZE,
+    OID_GEN_VENDOR_ID,
+    OID_GEN_VENDOR_DESCRIPTION,
+    OID_GEN_VENDOR_DRIVER_VERSION,
+    OID_GEN_CURRENT_LOOKAHEAD,
+    OID_802_3_MULTICAST_LIST,
+    OID_GEN_DRIVER_VERSION,
+    OID_GEN_MAXIMUM_TOTAL_SIZE,
+    OID_GEN_MAC_OPTIONS,
+    OID_GEN_MEDIA_CONNECT_STATUS,
+    OID_802_3_PERMANENT_ADDRESS,
+    OID_802_3_CURRENT_ADDRESS,
+    OID_802_3_MAXIMUM_LIST_SIZE,
+    /* Statistics */
+    OID_GEN_XMIT_OK,
+    OID_GEN_RCV_OK,
+    OID_GEN_XMIT_ERROR,
+    OID_GEN_RCV_ERROR,
+    OID_GEN_RCV_NO_BUFFER,
+};
+
+
+NDIS_STATUS
+NTAPI
+MiniportQueryInformation(
+    IN NDIS_HANDLE MiniportAdapterContext,
+    IN NDIS_OID Oid,
+    IN PVOID InformationBuffer,
+    IN ULONG InformationBufferLength,
+    OUT PULONG BytesWritten,
+    OUT PULONG BytesNeeded)
+{
+    PE1000_ADAPTER Adapter = (PE1000_ADAPTER)MiniportAdapterContext;
+    ULONG genericUlong;
+    ULONG copyLength;
+    PVOID copySource;
+    NDIS_STATUS status;
+
+    status = NDIS_STATUS_SUCCESS;
+    copySource = &genericUlong;
+    copyLength = sizeof(ULONG);
+
+    NdisDprAcquireSpinLock(&Adapter->AdapterLock);;
```

Comment:
Double semi-colon?

Reply:
Lol how does that compile? :D

Reply:
That's valid C : the second semi-colon corresponds to an empty instruction.